### ISSUE TITLE
improve the check_compatible function

### ISSUE THREAD

**Issue by qzhu2017** [qzhu2017/crystallography#6]
parameter set: 218 ['Ca', 'Li'] [4, 8] 1.0

If space group 218 has only 2a 6b as follows,
6 | b | 222 . . | (0,1/2,1/2)(1/2,0,1/2)(1/2,1/2,0)(0,1/2,0)(1/2,0,0)(0,0,1/2) | (0,1/2,1/2) | (1/2,0,1/2) | 
2 | a | 23. | (0,0,0)(1/2,1/2,1/2) | (0,0,0) | 

It is obviously impossible to make 4 atoms.

**Comment by qzhu2017** [comment]
Seems not a critical issue since the code is quite efficient now.

**Comment by scottfredericks** [comment]
The code now detects whether numIons is incompatible with a spacegroup. However, it is still possible for a generated crystal not to meet distance requirements due to the Wyckoff positions.
For example, spacegroup 226, with 32 atoms in its conventional cell, is forced into one of four configurations:
[24d and 8b], [24d and 8a], [24c and 8b], or [24c and 8a]
However, a crystal generated with any of these configurations will have inter-atomic distances which are too small. As a result, the program spends a long time trying to generate a crystal, but ultimately fails. For now, the code can tell if the valid Wyckoff positions have at least one degree of freedom (though we can easily alter it to tell whether there are 1, 2, or 3 degrees of freedom). I propose we use this information to change the method of generation. If there are no degrees of freedom, random crystal generation reduces to choosing a set of Wyckoff positions, and seeing if the choices work for any valid lattice. Alternatively, we could increase the volume factor for these cases, and then optimize the resulting structures later.

**Comment by scottfredericks** [comment]
I have found and fixed a bug for the parameter set (202, [20,3,8], [4,8, 10], 1.0). Previously, check_compatible simply ensured that each value of numIon is compatible with the space group. However, if atoms of one specie occupy a Wyckoff position with no degrees of freedom (for example, 1/2,1/2,1/2 or 1/4,3/4,3/4), atoms of a different specie should not be able to occupy the same position. To fix this, I added an array which stores already-filled Wyckoff positions. If a position is found in this array, the current atomic specie cannot use it.
